src/urllib3/response.py

```diff
@@ -869,6 +869,11 @@ def _raw_read(
                     # raised during streaming, so all calls with incorrect
                     # Content-Length are caught.
                     raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
+            elif read1 and (not data or self.length_remaining == len(data)):
+                # All data is read, but CPython's `self._fp.read1` doesn't
+                # always close `http.client.HTTPResponse`, so we close it
+                # here.
+                self._fp.close()
```

Comment:
Can you please add a link to the CPython issue and mention that this branch can be removed when we drop support for Python 3.12?

Reply:
Great idea, added